python-sdk/tests_integration/databases/test_mysql.py

```diff
@@ -0,0 +1,132 @@
+
+import pandas as pd
+import pytest
+import sqlalchemy
+
+from astro.constants import Database
+from astro.databases import create_database
+from astro.databases.mysql import MysqlDatabase
+from astro.settings import SCHEMA
+from astro.table import Metadata, Table
+
+DEFAULT_CONN_ID = "mysql_default"
+CUSTOM_CONN_ID = "mysql_conn"
+SUPPORTED_CONN_IDS = [DEFAULT_CONN_ID, CUSTOM_CONN_ID]
+CWD = pathlib.Path(__file__).parent
+
+TEST_TABLE = Table()
+
+
+@pytest.mark.parametrize("conn_id", SUPPORTED_CONN_IDS)
+def test_create_database(conn_id):
+    """Test creation of database"""
+    database = create_database(conn_id)
+    assert isinstance(database, MysqlDatabase)
+
+
+@pytest.mark.parametrize(
+    "conn_id,expected_uri",
+    [
+        (DEFAULT_CONN_ID, re.compile(r"^(mysql://)(.*)$")),
+        (
+            CUSTOM_CONN_ID,
+            r"^(mysql://)(.*:3306/)(astrodb|tmp_astro)$",
+        ),
+    ],
+    ids=SUPPORTED_CONN_IDS,
+)
+def test_mysql_sqlalchemy_engine(conn_id, expected_uri):
+    """Test getting a mysql based sqla engine."""
+    database = MysqlDatabase(conn_id)
+    engine = database.sqlalchemy_engine
+    assert isinstance(engine, sqlalchemy.engine.base.Engine)
+    url = urlparse(str(engine.url))
+    assert re.match(expected_uri, url.geturl())
+
+
+@pytest.mark.integration
+def test_mysql_run_sql():
+    """Test run_sql against mysql database"""
+    statement = "SELECT 1 + 1;"
+    database = MysqlDatabase(conn_id=CUSTOM_CONN_ID)
+    response = database.run_sql(statement, handler=lambda x: x.first())
+    assert response[0] == 2
+
+
+@pytest.mark.integration
+def test_table_exists_raises_exception():
+    """Test if table exists in mysql database"""
+    database = MysqlDatabase(conn_id=CUSTOM_CONN_ID)
+    table = Table(name="non-existing-table", metadata=Metadata(schema=SCHEMA))
+    assert not database.table_exists(table)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.MYSQL,
+            "table": Table(
+                metadata=Metadata(database=SCHEMA, schema=SCHEMA),
+                columns=[
+                    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+                    sqlalchemy.Column("name", sqlalchemy.String(60), nullable=False, key="name"),
+                ],
+            ),
+        }
+    ],
+    indirect=True,
+    ids=["mysql"],
+)
+def test_mysql_create_table_with_columns(database_table_fixture):
+    """Test table creation with columns data"""
+    database, table = database_table_fixture
+
+    statement = f"SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name='{table.name}'"
+    response = database.run_sql(statement, handler=lambda x: x.first())
+    assert response is None
+
+    database.create_table(table)
+    response = database.run_sql(statement, handler=lambda x: x.fetchall())
+    rows = response
+    assert len(rows) == 2
+    assert rows[0][1:4] == (
```

Comment:
I think we can make this test more robust, right now if the order of keys change from DB this might fail. We can maybe pass column names in SQL statement 
 `f"SELECT <col_1>, <col_2> ,... FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name='{table.name}'"`

Reply:
its very rare that a system table column order will get changed, however I've changed the query to use column names as suggested